scripts/typescript-compatibility-build.yml

```diff
@@ -0,0 +1,120 @@
+trigger:
+  branches:
+    include:
+      - tsc-compat-debug
+      #- refs/tags/office-ui-fabric-react_v*
+
+variables:
+  - name: PackageName
+    value: office-ui-fabric-react
+  - name: CompatRepoUrl
+    value: https://github.com/Microsoft/ui-fabric-ts-validation.git
+  - name: NodeVersion
+    value: '10.x'
+  - name: VmImage
+    value: 'Ubuntu 16.04'
+
+jobs:
+  - job: Build
+    pool:
+      vmImage: $(VmImage)
+    timeoutInMinutes: 60
+    steps:
+      - checkout: self
+        clean: true
+
+      - task: NodeTool@0
+        displayName: 'Use Node $(NodeVersion)'
+        inputs:
+          versionSpec: '$(NodeVersion)'
+          checkLatest: true
+
+      # Install Rush repo dependencies
+      - script: |
+          node common/scripts/install-run-rush.js install --bypass-policy
+        displayName: Run Rush install in office-ui-fabric-react repository
+
+      # Build package in Rush repo
+      - script: |
+          node common/scripts/install-run-rush.js build --to $(PackageName)
+        displayName: Run Rush build --to $(PackageName)
+
+      # Run npm pack to simulate installation from feed
+      - script: |
+          npm pack ./packages/$(PackageName)
+        displayName: Run npm pack $(PackageName)
+
+      # Rename built package tarball to reference later
+      - script: |
+          mv *.tgz $(Build.StagingDirectory)/$(PackageName).tgz
+        displayName: Move $(PackageName).tgz to Build.StagingDirectory
+
+      # Publish built library for validation job
+      - task: PublishBuildArtifacts@1
+        displayName: Publish Build Artifacts
+        inputs:
+          pathtoPublish: $(Build.StagingDirectory)
+          artifactName: drop
+
+  - job: Compatibility
+    dependsOn: Build
+    condition: succeeded()
+    timeoutInMinutes: 60
+    pool:
+      vmImage: $(VmImage)
+    strategy:
+      maxParallel: 2
+      matrix:
+        'TypeScript 2.4':
```

Comment:
Do we expect current OUFR to build against 2.4 and 2.7? Especially for 2.4, I'd expect a huge number of breaks.

Reply:
Correct, OUFR >= `6.x` does not build against `2.4`. @aditima and I think that this is useful information to surface as part of the build. One of the friction points today is that SPFx developers use various versions of the compiler, `2.4` being one of them ([rush-stack-compiler releases](https://github.com/microsoft/web-build-tools/tree/master/stack)).

We want to be able to direct folks to this running build as a way for them to decide what version of TypeScript they need to upgrade to in order to be compatible. @aditima had a good point that if we only surface passing builds then folks would still be unsure what versions may lead to problems.

We'll figure out how to handle the build alerting spam that comes along with this decision as part of wiring that up.